nautobot/extras/models/jobs.py

```diff
@@ -799,6 +814,80 @@ def enqueue_job(
         if task_queue is None:
             task_queue = job_model.default_job_queue.name
 
+        job_queue = JobQueue.objects.get(name=task_queue)
+        # Kubernetes Job Queue logic
+        if job_queue.queue_type == JobQueueTypeChoices.TYPE_KUBERNETES:
+            kube_config_context = job_queue.context
```

Comment:
Maybe split all of this out into a private helper function? `enqueue_job()` is already beastly long.